jupyterhub/handlers/pages.py

```diff
@@ -24,6 +27,11 @@ class RootHandler(BaseHandler):
     Otherwise, renders login page.
     """
     def get(self):
+        next_url = self.get_argument('next', '')
+        if next_url and next_url.startswith('/user/'):
```

Comment:
`startswith('/user/')` won't handle JupyterHub on a prefix. I think this needs to be `url_path_join(self.base_url, 'user') + '/'`